cmd/influx/user_resource_mapping.go

```diff
@@ -3,29 +3,96 @@ package main
 import (
 	"context"
 	"fmt"
+	"os"
 
 	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/cmd/influx/internal"
 )
 
-func membersAddF(urm influxdb.UserResourceMapping, flgs Flags) error {
+func membersListF(ctx context.Context, filter influxdb.UserResourceMappingFilter, flgs Flags) error {
+	mappingSvc, err := newUserResourceMappingService(flgs)
+	if err != nil {
+		return fmt.Errorf("failed to initialize members service client: %v", err)
+	}
+	mps, _, err := mappingSvc.FindUserResourceMappings(ctx, filter)
+	if err != nil {
+		return fmt.Errorf("failed to find members: %v", err)
+	}
+	userSVC, err := newUserService(flgs)
+	if err != nil {
+		return fmt.Errorf("failed to initialize users service client: %v", err)
+	}
+	urs := make([]*influxdb.User, len(mps))
+	ursC := make(chan struct {
+		User  *influxdb.User
+		Index int
+	})
+	errC := make(chan error)
+	for k, v := range mps {
+		go func(k int, v *influxdb.UserResourceMapping) {
+			usr, err := userSVC.FindUserByID(ctx, v.UserID)
+			if err != nil {
+				errC <- fmt.Errorf("failed to retrieve user details: %v", err)
+				return
+			}
+			ursC <- struct {
+				User  *influxdb.User
+				Index int
+			}{
+				User:  usr,
+				Index: k,
+			}
+		}(k, v)
+	}
+	for i := 0; i < len(mps); i++ {
+		select {
+		case <-ctx.Done():
```

Comment:
this done won't get called until the background context is canceled yah? Perhaps a `context.WithTimeout(ctx, 10*time.second)` or something would suffice here?  You could also use `time.After(10*time.Second)`, or both depending on how you want to signal cancelation from the OS

Reply:
yeah, i want to handle the timeout in other ticket. so it won't just change here.
https://github.com/influxdata/influxdb/issues/15375

Reply:
right on sounds good 👍